Automattic/wp-calypso
# A4A: Add Marketplace dedicated sidebar section.

This pull request adds a new Marketplace sidebar section to match the updated Information Architecture. The updated architecture now has a dedicated Products page and Hosting page.
<img width="1610" alt="Screenshot 2024-03-14 at 4 15 39 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/739efc49-b449-47a6-b845-3eb18fcff3c6">
<img width="1614" alt="Screenshot 2024-03-14 at 4 15 47 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/8466cc48-8a94-4c25-93d5-055fc98f9f64">

Closes https://github.com/Automattic/jetpack-genesis/issues/282

## Proposed Changes

* Add Marketplace sidebar navigation menu.
* Add Hosting overview placeholder page.
* Update Marketplace controller to include 'Products' and 'Hosting' routes.
* Update Layout to support spacing with top border. Additionally, I added new 'compact' props for different spacing sizes. This will be used later on the checkout and Pressable/Wordpress.com site page.

## Testing Instructions
* Switch branch: `git checkout add/a4a-marketplace-sidebar-section`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/
* Click the 'Marketplace' sidebar menu item.
* Confirm that you are redirected to the Products page. 
* Click 'Hosting' on the sidebar menu item.
* Confirm that you are redirected to the Hosting page.

## Pre-merge Checklist
